express-notes-api/server.js

```diff
@@ -0,0 +1,113 @@
+import express from 'express';
+import { readFile, writeFile } from 'fs/promises';
+
+const app = express();
+app.use(express.json());
+
+app.get('/api/notes/:id', async (req, res) => {
+  // noteId holds the id that is a key in data.json
+  const noteId = req.params.id;
+
+  try {
+    // Reading the JSON file and waiting for its response
+    const data = JSON.parse(await readFile('./data.json', 'utf-8'));
```

Comment:
It would be nice to encapsulate this into a function so you don't have to write it over and over.